wwwapp/models.py

```diff
@@ -60,6 +60,48 @@ def __str__(self):
         return '%s: %s, %s' % (self.year, self.user_profile, self.status)
 
 
+class PESELField(models.CharField):
+    """PESEL field, with checksum verification."""
+
+    def validate(self, pesel: str, user_info: 'UserInfo') -> None:
+        super().validate(pesel, user_info)
+        # We accept empty PESEL (don't raise exception) for legacy reasons.
+        if not pesel:
+            return
+
+        # https://en.wikipedia.org/wiki/PESEL#Format
+        if len(pesel) != 11:
+            raise ValidationError('PESEL length is wrong ({})'.format(len(pesel)))
+        if not pesel.isdigit():
+            raise ValidationError('PESEL contains non-numbers')
+
+        pesel_digits = [int(digit) for digit in pesel]
+        checksum_mults = [1, 3, 7, 9] * 2 + [1, 3, 1]
+        if sum(x*y for x, y in zip(pesel_digits, checksum_mults)) % 10 != 0:
+            raise ValidationError('PESEL checksum is not valid')
```

Comment:
We shouldn't throw exception here as some people may have badly formed PESELs

Reply:
We already discussed this at the very beginning, @cytadela8 said that all such mistakes were fixed a long time ago

Reply:
Still, I wouldn't risk. We can validate it on the frontend and display warning about it

Reply:
@radrow These mistakes were fixed long ago. Django does some magic in order for the ValidationError to show up in the form - the user will see the error message.

Reply:
Your will, I won't defend you in the lawsuit

Reply:
@radrow in that case one just needs to adjust last digit so it passes the checksum test xd.

Reply:
We'll worry about it if it happens :P